lib/hydra/works/models/concerns/collection_behavior.rb

```diff
@@ -22,6 +22,39 @@ module CollectionBehavior
       type [RDFVocabularies::PCDMTerms.Collection,WorksVocabularies::WorksTerms.Collection]
       include Hydra::Works::AggregatesGenericWorks
       include Hydra::Works::AggregatesCollections
+      include Hydra::Works::BlockChildObjects
+
+      filters_association :members, as: :child_collections, condition: :works_collection?
+      filters_association :members, as: :child_generic_works, condition: :works_generic_work?
+    end
+
+    module ClassMethods
+      def indexer
+        Hydra::Works::CollectionIndexer
+      end
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Collection.
+    def works_collection?
+      true
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic Work.
+    def works_generic_work?
+      false
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic File.
+    def works_generic_file?
```

Comment:
```ruby
generic_work.child_generic_files += [col1]
```
This will call col1.works_generic_file? and return false which will cause the filter association code to print out a nice message for the caller telling them they passed col1 which is the wrong type.